File: 03-recursion/ruby/recursion_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative './recursion'

class RecursionTest < Minitest::Test
  def test_factorial_returns_an_integer
    assert_instance_of Integer, factorial(3)
  end

  def test_factorial_returns_base_case
    assert_equal 1, factorial(1)
  end

  def test_factorial_returns_correct_result
    assert_equal 120, factorial(5)
  end
end

File: 03-recursion/python/recursion_test.py
import unittest
from .recursion import factorial

class TestRecursion(unittest.TestCase):
    def Setup(self):
        pass

    def test_factorial_returns_a_number(self):
        self.assertIsInstance(factorial(3), int)

    def test_factorial_returns_base_case(self):
        self.assertEqual(factorial(1), 1)

    def test_factorial_returns_correct_result(self):
        self.assertEqual(factorial(5), 120)
